File: Backend/HoPla_API/Entities/House.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace HoPla_API.Entities
{
    public class House
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; }

        [Required]
        public bool HasPremium { get; set; }

        public int? HouseSize { get; set; }

        [AllowNull]
        public ICollection<Item> Items { get; set; }

        public House(string name, bool hasPremium)
        {
            Name = name;
            HasPremium = hasPremium;
            Items = new List<Item>();
        }
    }

    public class HouseInputModel
    {
        [MaxLength(50)]
        public string Name { get; set; }

        public bool HasPremium { get; set; }

        public int? HouseSize { get; set; }
    }
}

File: Frontend/lib/model/house.dart
import 'package:json_annotation/json_annotation.dart';

import 'item.dart';

part 'house.g.dart';

@JsonSerializable()
class House {
  int id;
  String name;
  bool hasPremium;
  int houseSize;
  List<Item> items;

  House({
    required this.id,
    required this.name,
    required this.hasPremium,
    required this.houseSize,
    required this.items,
  });


  factory House.fromJson(Map<String, dynamic> json) => _$HouseFromJson(json);

  Map<String, dynamic> toJson() => _$HouseToJson(this);
}
